src/contrib/cluster/Akka.Cluster.Sharding.Tests/ClusterShardingMessageSerializerSpec.cs

```diff
@@ -140,12 +176,40 @@ public void ClusterShardingMessageSerializer_must_be_able_to_serialize_StartEnti
             CheckSerialization(new ShardRegion.StartEntityAck("13", "37"));
         }
 
+        [Fact]
+        public void ClusterShardingMessageSerializer_must_be_able_to_serialize_GetCurrentRegions()
+        {
+            CheckSerialization(GetCurrentRegions.Instance);
+            CheckSerialization(new CurrentRegions(ImmutableHashSet.Create(new Address("akka", "sys", "a", 2552), new Address("akka", "sys", "b", 2552))));
+        }
+
         [Fact]
         public void ClusterShardingMessageSerializer_must_serialize_ClusterShardingStats()
         {
-            CheckSerialization(new GetClusterShardingStats(TimeSpan.FromMilliseconds(500)));
-            CheckSerialization(new ClusterShardingStats(ImmutableDictionary<Address, ShardRegionStats>.Empty.Add(new Address("akka.tcp", "foo", "localhost", 9110), 
-                new ShardRegionStats(ImmutableDictionary<string, int>.Empty.Add("f", 12)))));
+            CheckSerialization(new GetClusterShardingStats(TimeSpan.FromSeconds(3)));
+
+            CheckSerialization(new ClusterShardingStats(ImmutableDictionary<Address, ShardRegionStats>.Empty
+                .Add(new Address("akka.tcp", "sys", "a", 2552), new ShardRegionStats(ImmutableDictionary<string, int>.Empty.Add("a", 23), ImmutableHashSet.Create("b")))
+                .Add(new Address("akka.tcp", "sys", "b", 2552), new ShardRegionStats(ImmutableDictionary<string, int>.Empty.Add("a", 23), ImmutableHashSet.Create("b")))
+                ));
+        }
+
+        [Fact]
+        public void ClusterShardingMessageSerializer_must_serialize_ShardQuery()
+        {
+            CheckSerialization(Shard.GetCurrentShardState.Instance);
+            CheckSerialization(new Shard.CurrentShardState("12", ImmutableHashSet.Create("a", "b")));
+        }
+
+        [Fact]
+        public void ClusterShardingMessageSerializer_must_serialize_ShardRegionQuery()
```

Comment:
LGTM - all of the new tests in this class look fine, but I haven't checked the underlying serializer impl itself which is where I know @Arkatufus raised some points a while back.